client/resource_group/controller/controller.go

```diff
@@ -469,6 +474,9 @@ func (c *ResourceGroupsController) OnResponse(
 		log.Warn("[resource group controller] resource group name does not exist", zap.String("name", resourceGroupName))
 		return &rmpb.Consumption{}, nil
 	}
+	if tmp.(*groupCostController).isBackgroundRequest(req.RequestSource()) {
```

Comment:
Maybe we label the req as background before `OnRequestWait`? then we can directly judge it by `RequestInfo`. like adding an interface about judge background first, and setting the `RequestInfo` as background, then `OnRequestWait`, `OnResponds`, or others can share the result.

Reply:
You can pass settings to client-go?

Reply:
Let the client-go have a way to know the type (background/foreground) of the request. and then we no need to check inside the `OnResponds`, we an directly bypass `OnResponds`.

Reply:
done. add `IsBackgroundRequest` to check in client-go, PTAL, thx!
https://github.com/tikv/client-go/pull/872/files#diff-34812587e42268a5d84933be286601fc58fdff17ab336b09ba75455ffe5ed447R111